### PR TITLE
Remove unused local variables

### PR DESCRIPTION
**Pull request by pzygielo**

### CODE REVIEW

**Comment by bshannon** [comment]
@pzygielo are you just exercising your IDE and finding all this trivial "problems"?
Since I'm busy with many things and trying to get this release stabilized and complete,
I'm not at all convinced it's worth my time to analyze every one of these code "improvements".
I'm likely to leave these to a future release when I have more time.

**Comment by pzygielo** [comment]
@bshannon 
> @pzygielo are you just exercising your IDE and finding all this trivial "problems"?

Regardless of tool used - they are true cases.

> I'm likely to leave these to a future release when I have more time.

Seems for so many years there was only time to introduce them.

**Comment by bshannon** [comment]
> Seems for so many years there was only time to introduce them.

That's exactly right.  In 20 years we had time to introduce 100's of features and
bug fixes but we only had time to introduce 6 unused local variables!  We'll try
harder the next 20 years.

**Comment by pzygielo** [comment]
> but we only had time to introduce 6 unused local variables! 

That's why I donated my time to make it even less.

---

True - I raised several PRs for - as you found them - _trivial problems_. Sometimes they are called code smells or by other names. They interfere with (at least) my reception of the rest of the code. I assumed it would be welcomed change - easy to review and non-controversial. No API changes and no conflicts between separate PRs.

And I was waiting for 2.0.0 being merged to master. That happened - so I PRed my changes.

I waited for Jenkins to run (twice a day, not on PR creation) and checked build status to confirm that it failed not on build but on TCK before marking PR as ready for review. To not waste your time too early. You decided to comment before that.

> I'm likely to leave these to a future release

I've already learnt that JavaEE life cycle is kind of extremely long.

> when I have more time.

But https://projects.eclipse.org/projects/ee4j.mail/who lists many Committers that Eclipse trusts as well.

**Comment by bshannon** [comment]
Everything takes time, and it's my time to review the changes that are what's critical here.
Yes, there are other committers, but none of them have touched, or probably even read,
the relevant code.  They would have no idea why an unused variable may have been left
on purpose.

Your changes are not unwelcome, it's just the timing that is bad.  Of all the things we need
help with  across all the Jakarta EE projects for Jakarta EE 9, finding unused variables is not
a high priority.  Once we get this first critical release done, we'll have more breathing room
to deal with these code quality concerns.

And I think one of the bigger points here is that it's always better to ask before contributing
your time.  Had you showed up on the dev list and said "I can contribute a PR that finds all
the unused local variables, would that help?" my answer would have been "now is not the
best time for such changes, I'm trying to stabilize the code and manage changes across
two branches so the fewer unnecessary changes the better."

To be clear, once this release is done, I'll look at your PR and incorporate it as appropriate.
Now is just not the best time.

**Comment by pzygielo** [comment]
>  Had you showed up on the dev list 

No, I had not. :cry: 
But I read https://www.eclipse.org/lists/javamail-dev/msg00027.html:

> When that's done, I'll merge the existing 2.0.0 branch, containing the
change to the jakarta.mail package namespace, into the master branch.
At that point it's mostly a matter of integrating it with GlassFish for
the Jakarta EE 9 release.

and I waited for that to happen.

> Now is just not the best time.

I'm fine with that.

File: mail/src/main/java/com/sun/mail/imap/protocol/IMAPProtocol.java [hunk 1]
```diff
@@ -761,7 +761,6 @@ public synchronized void authntlm(String authzid, String u, String p)
 	Response r = null;
 	boolean done = false;
 
-	String type1Msg = null;
```
Comment (Cousjava) [inline mail/src/main/java/com/sun/mail/imap/protocol/IMAPProtocol.java]: This became used at https://github.com/javaee/javamail/commit/081f632e1fd8ae5fa6f96a0360dc9a681c3eaa79

File: mail/src/main/java/com/sun/mail/imap/protocol/BODYSTRUCTURE.java [hunk 2]
```diff
@@ -81,7 +81,6 @@ public BODYSTRUCTURE(FetchResponse r) throws ParsingException {
 	    type = "multipart";
 	    processedType = MULTI;
 	    List<BODYSTRUCTURE> v = new ArrayList<>(1);
-	    int i = 1;
```
Comment (Cousjava) [inline mail/src/main/java/com/sun/mail/imap/protocol/BODYSTRUCTURE.java]: .Was unused at point git history started

File: mail/src/main/java/com/sun/mail/imap/protocol/UIDSet.java [hunk 3]
```diff
@@ -86,7 +86,6 @@ public long size() {
 	    return null;
 	List<UIDSet> v = new ArrayList<>();
 	StringTokenizer st = new StringTokenizer(uids, ",:", true);
-	long start = -1;
```
Comment (Cousjava) [inline mail/src/main/java/com/sun/mail/imap/protocol/UIDSet.java]: Unused in commit introduced in https://github.com/javaee/javamail/commit/2bc042e76d63147093c80189c0d9aeab8c49f789

File: mail/src/main/java/com/sun/mail/pop3/POP3Message.java [hunk 4]
```diff
@@ -179,7 +179,6 @@ private InputStream getRawStream(boolean skipHeader)
 		     * the content, skipping over the header until we come to
 		     * the empty line that separates the header from the body.
 		     */
-		    int offset = 0;
```
Comment (Cousjava) [inline mail/src/main/java/com/sun/mail/pop3/POP3Message.java]: Unused in initial git commit in Java EE

File: mail/src/main/java/com/sun/mail/smtp/DigestMD5.java [hunk 5]
```diff
@@ -77,7 +77,6 @@ public DigestMD5(MailLogger logger) {
 	String nc = "00000001";
 	String qop = "auth";
 	byte[] bytes = new byte[32];	// arbitrary size ...
-	int resp;
```
Comment (Cousjava) [inline mail/src/main/java/com/sun/mail/smtp/DigestMD5.java]: Unused in initial git commit

File: mail/src/main/java/jakarta/mail/internet/MimeMessage.java [hunk 6]
```diff
@@ -1716,7 +1716,6 @@ public Message reply(boolean replyToAll, boolean setAnswered)
 		eliminateDuplicates(v,
 				InternetAddress.parse(alternates, false));
 	    // should we Cc all other original recipients?
-	    String replyallccStr = null;
```
Comment (Cousjava) [inline mail/src/main/java/jakarta/mail/internet/MimeMessage.java]: Became unused in https://github.com/javaee/javamail/commit/d9b77024e98795a753e3d95f29164112579d5e3a

**Comment by bshannon** [comment]
> As one of the other mail committers, I have looked through the history for all of them and found where they became unused, and I don't see any point in keeping them.

Thanks for checking.  I'll consider it for the next release.

**Comment by pzygielo** [comment]
Drafted to rebase onto #452

**Comment by pzygielo** [comment]
Unfortunately master is not-green again as well:
mail-master
:disappointed:

**Pull request merged**